draft-tjjk-cared.md

```diff
@@ -1,336 +1,405 @@
+protocols that incorporate TLS or dTLS such as DoH {{?RFC8484}},
+DoT {{?RFC7858}}, and DoQ {{?RFC9250}} SHOULD NOT provide client
+authentication except in the limited situations described in this
+section.
+
+Encrypted DNS servers that use other transport protocols are not
+in scope for this document.
+
+Encrypted DNS servers MUST NOT challenge clients for authentication unless they need to
+restrict connections to a set of clients they have a pre-existing relationship with as defined
+in {{#restrict-clients}}, regardless of whether or not the requirements in {{per-client}} apply.
+Encrypted DNS servers that meet the requirements in {{restrict-clients}} MAY
+challenge clients to authenticate to avoid achieving the same goal of identifying clients
+through other, less secure means (such as IP address or data in the DNS query payload).
+
+### Restricting connections to allowed clients {#restrict-clients}
+
+An encrypted DNS server that provides resolution to a specific set
+of clients and refuses service to all other clients MAY require
+clients to authenticate.
+
+For example, an encrypted DNS server owned by an enterprise that
+only allows connections from devices managed by that same enterprise
+might require clients to authenticate.
+
+### Resolving names differently per client {#per-client}
+
+An encrypted DNS server that provides client-specific resolution
+behaviour MAY require clients to authenticate in circumstances
+where the appropriate policy for particular clients is
+specified by the operator of the encrypted DNS server.
+
+For example, an encrypted DNS server that is configured to allow
+some clients to resolve certain names while other clients are not
+allowed needs to identify the client so that the resolution behaviour
+for those names can be implemented accurately.
+
+Encrypted DNS servers SHOULD NOT attempt to authenticate clients
+to identify the appropriate resolution policy to use when the
+difference in resolution behavior between them is not imposed by
+the operator of the server but instead is chosen by the client.
+
+For example, some public encrypted DNS services provide clients
+with the option of blocking resolution requests for particular
+categories of domain names, such as names associated with malware
+distribution, adult content or advertisers, and makes servers with
+particular resolution policies available on different IP addresses.
+A client who wishes to opt-in to a server with a particular resolution
+policy can do so by sending queries to the corresponding IP address,
+and no client identification is required.
+
+## When clients should attempt to authenticate
+
+An encrypted DNS client MUST NOT offer authentication to any encrypted
+DNS server unless it was specifically configured to expect that
+server to require authentication, independent of the mechanism by
+which the client chose or discovered the encrypted DNS server to
+use.
+
+An encrypted DNS client MUST NOT offer authentication when the
```

Comment:
We could add the missing "SHOULD NOT when the server has no strong identity" bit here as the client MUST NOT when using opportunistic DoT for a more natural flow (see other comment where this normative was removed)

Reply:
Are those cases not already covered explicitly by the MUST NOT in this paragraph? Wouldn't a SHOULD NOT for particular examples of the client not having explicit configuration for a server act as a downgrade?

Reply:
No, because DDR is not the only way the server may know this no-name-to-verify will occur, such as opportunistic DoT. From the server's perspective, it doesn't know if the client is being opportunistic of course, but it will know the client has little choice if the server knows it has a self-signed cert.

Per your comment, I will take a stab at this (but in a new PR, I don't want to keep holding this up).